stock_reserve_rule/models/stock_reserve_rule.py

```diff
@@ -296,3 +313,37 @@ def is_greater_eq(value, other):
                     # compute how much packaging we can get
                     take = (need // pack_quantity) * pack_quantity
                     need = yield location, location_quantity, take, None, None
+
+    def _apply_strategy_by_lot(self, quants):
+        need = yield
+        product = fields.first(quants).product_id
+        comparison = {
+            "lot_qty_gt_reserved_qty": lambda lot_qty, need_qty: lot_qty > need_qty,
+            "lot_qty_gte_reserved_qty": lambda lot_qty, need_qty: lot_qty >= need_qty,
+            "lot_qty_eq_reserved_qty": lambda lot_qty, need_qty: lot_qty == need_qty,
+        }.get(self.reserve_lot)
+        if product.tracking == "lot" or comparison is not None:
+            for location, location_quants in quants._group_by_location():
+                grouped_quants = self.env["stock.quant"].read_group(
+                    [
+                        ("lot_id", "!=", False),
+                        ("location_id", "in", location.ids),
+                        ("product_id", "=", product.id),
+                    ],
+                    ["lot_id", "quantity"],
+                    ["lot_id"],
+                    orderby="id",
+                )
+                lot_ids_with_quantity = {
+                    group["lot_id"][0]: group["quantity"] for group in grouped_quants
+                }
+                location_quantity = sum(location_quants.mapped("quantity")) - sum(
+                    location_quants.mapped("reserved_quantity")
+                )
+                lot_id = False
+                for rec_id, product_qty in lot_ids_with_quantity.items():
+                    if comparison(product_qty, need):
+                        lot_id = rec_id
```

Comment:
You need to respect default removal strategy (FIFO, FEFO, ...)

Reply:
> You need to respect default removal strategy (FIFO, FEFO, ...)

What do you mean?

Reply:
> > You need to respect default removal strategy (FIFO, FEFO, ...)
> 
> What do you mean?

@jbaudoux any suggestions ?

Reply:
@geomer198 please check it

Reply:
@geomer198 any news ?

Reply:
@jbaudoux I think use `quants._group_by_lot()` not solution because strategy need return `location.id`. This function return `lot` record and `quants` records that have many locations maybe.  That is why I use `group_by_location` and than do read_group for search `quants` by location and product.

Reply:
You could change what the method returns to provide the arguments required by `_update_reserved_quantity`.
You could even return a list and, in the caller, loop on the results to call `_update_reserved_quantity` on each element.
You anyway need to return the lot to ensure you reserve that selected lot in case there are multiple lots on the location.

Reply:
@geomer198 what about this?